### PR TITLE
Add aggregators for double & long MinMaxSumCount aggregations

### PR DESCRIPTION
**Pull request by jkwatson**
And, simple benchmarks for them.

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  public static AggregatorFactory getFactory() {
+    return AGGREGATOR_FACTORY;
+  }
+
+  @Override
+  void doMergeAndReset(Aggregator target) {
+    DoubleMinMaxSumCount other = (DoubleMinMaxSumCount) target;
+
+    DoubleSummary copy = current.copyAndReset();
+    other.current.update(copy);
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java]: what about just:
`current.mergeAndReset(other.current)`

```java
    private void mergeAndReset(DoubleSummary target) {
      long count;
      double sum;
      double min;
      double max;
      lock.writeLock().lock();
      try {
        count = this.count;
        sum = this.sum;
        min = this.min;
        max = this.max;
        this.count = 0;
        this.sum = 0;
        this.min = 0;
        this.max = 0;
      } finally {
        lock.writeLock().unlock();
      }
      target.update(count, sum, min, max);
    }

    private void update(long count, double sum, double min, double max) {
      if (count == 0) {
        return;  // nothing to do
      }
      lock.writeLock().lock();
      try {
        this.count += count;
        this.sum += sum;
        this.min = Math.min(min, this.min);
        this.max = Math.max(max, this.max);
      } finally {
        lock.writeLock().unlock();
      }
    }
```

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        long startEpochNanos, long epochNanos, Map<String, String> labels) {
+      lock.readLock().lock();
+      try {
+        return SummaryPoint.create(
+            startEpochNanos,
+            epochNanos,
+            labels,
+            count,
+            sum,
+            min == null || max == null // note: min & max will be null or not null together.
```
[the remark this answers is not in the document]
Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java]: excellent idea. will make it so.

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java (hunk 1 above)
Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java]: I like this idea. I'll try it out.

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java]: @jkwatson I am able to come with this ideas only if I have small PRs :). No seriously I would have not been able to see this in 1000 lines PR :)

[the remark this answers is not in the document]
Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java]: It's all good. Sometimes I need to build the full solution to make sure I understand it; I'm happy to go back and PR in chunks after the fact.

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        long startEpochNanos, long epochNanos, Map<String, String> labels) {
+      lock.readLock().lock();
+      try {
+        return SummaryPoint.create(
+            startEpochNanos,
+            epochNanos,
+            labels,
+            count,
+            sum,
+            count == 0
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java]: Return null if no values?

Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java]: yes yes yes. not sure why I didn't do that the first time around.

File: sdk/src/jmh/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCountBenchmark.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import org.openjdk.jmh.annotations.Warmup;
+
+@State(Scope.Benchmark)
+public class DoubleMinMaxSumCountBenchmark {
+
+  private DoubleMinMaxSumCount aggregator;
+
+  @Setup(Level.Trial)
+  public final void setup() {
+    aggregator = new DoubleMinMaxSumCount();
```
Comment (bogdandrutu) [inline sdk/src/jmh/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCountBenchmark.java]: Get the instance via the factory and add a private ctor to the class, so to enforce usage of the factory :)

Comment (jkwatson) [reply sdk/src/jmh/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCountBenchmark.java]: sure thing.

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java]: This is the only issue left

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        mySum = this.sum;
+        myMin = this.min;
+        myMax = this.max;
+        this.count = 0;
+        this.sum = 0;
+        this.min = Long.MAX_VALUE;
+        this.max = Long.MIN_VALUE;
+      } finally {
+        lock.writeLock().unlock();
+      }
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java]: if this.count == 0 you can return immediately.

Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java]: true true

Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java]: also, the default values are wrong here.  fixing.

**Pull request merged**